crates/starknet/src/starknet/dump.rs

```diff
@@ -0,0 +1,187 @@
+                            tx.version,
+                        );
+                        self.add_declare_transaction_v1(declare_tx)?;
+                    } else {
+                        return Err(Error::SerializationNotSupported);
+                    };
+                }
+                Transaction::Declare(DeclareTransaction::Version2(tx)) => {
+                    let declare_tx = BroadcastedDeclareTransactionV2::new(
+                        &tx.contract_class,
+                        tx.compiled_class_hash,
+                        tx.sender_address,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        tx.version,
+                    );
+                    self.add_declare_transaction_v2(declare_tx)?;
+                }
+                Transaction::DeployAccount(tx) => {
+                    let deploy_account_tx = BroadcastedDeployAccountTransaction::new(
+                        &tx.constructor_calldata,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        tx.class_hash,
+                        tx.contract_address_salt,
+                        tx.version,
+                    );
+                    self.add_deploy_account_transaction(deploy_account_tx)?;
+                }
+                Transaction::Deploy(_) => return Err(Error::SerializationNotSupported),
+                Transaction::Invoke(InvokeTransaction::Version0(_)) => {
+                    return Err(Error::SerializationNotSupported);
+                }
+                Transaction::Invoke(InvokeTransaction::Version1(tx)) => {
+                    let invoke_tx = BroadcastedInvokeTransaction::new(
+                        tx.sender_address,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        &tx.calldata,
+                        tx.version,
+                    );
+                    self.add_invoke_transaction(invoke_tx)?;
+                }
+                Transaction::L1Handler(_) => return Err(Error::SerializationNotSupported),
+            };
+        }
+
+        Ok(())
+    }
+
+    /// attach starknet transaction to end of existing file
+    pub fn dump_transaction(&self, transaction: &Transaction) -> DevnetResult<()> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let file_path = Path::new(path);
+                if file_path.exists() {
+                    // attach to file
+                    let transaction_dump = serde_json::to_string(transaction).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    let mut file = OpenOptions::new()
+                        .append(true)
+                        .write(true)
+                        .read(true)
```

Comment:
I don't remember ever seeing a file being opened in all three modes (append, write, read). Are we sure we need all three of them?

Reply:
yes it was needed to add read

Reply:
I was also stunned

Reply:
but couldn't we just have append and read?

Reply:
write is needed to change size (remove last "]")

Reply:
I found this: https://stackoverflow.com/a/30684735
It says that since Rust 1.8 `.write(true).append(true)` is the same as `.append(true)`. And really, the docs say the same (we are using Rust 1.69).

Reply:
ok let me check it in code and tests

Reply:
Fixed in 1c5b6e17148162914e6c57101530caab7d288f66